Identify cause of connected test failures on Firebase Test Lab

The new Espresso tests added in [`FieldListUpdateTest`](https://github.com/opendatakit/collect/blob/b408e535d6ca2b767fff8c1549f5caa4ddaed935/collect_app/src/androidTest/java/org/odk/collect/android/formfilling/FieldListUpdateTest.java) appear to fail intermittently on Firebase Test Lab.

First I thought the issue was that when selecting a group from the hierarchy view we only scrolled until the item we were looking for appeared on the screen at all but when clicking we required 90% visibility. But https://github.com/opendatakit/collect/pull/3092 didn't fix the problem.

b408e535d6ca2b767fff8c1549f5caa4ddaed935 ignores the test that I thought was always failing but now `selectingADateForDateTime_ShouldChangeRelevanceOfRelatedField` which passed previously is failing instead.

Looking through the stacktrace below, I can see that the toplevel list of groups is being shown when I expect to be in the `Date time` group. I'm wondering whether it could be a timing issue even though Espresso is supposed to take care of that.

<details>
<summary>Stack trace</summary>
<pre>
androidx.test.espresso.NoMatchingViewException: No views in hierarchy found matching: with text: a string starting with "Source12"

View Hierarchy:
+>DecorView{id=-1, visibility=VISIBLE, width=1080, height=1920, has-focus=true, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=WM.LayoutParams{(0,0)(fillxfill) ty=1 fl=#81810100 pfl=0x8 wanim=0x1030461 surfaceInsets=Rect(0, 0 - 0, 0)}, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=3}
|
+->LinearLayout{id=-1, visibility=VISIBLE, width=1080, height=1776, has-focus=true, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@238d01b, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=2}
|
+-->ViewStub{id=16909166, visibility=GONE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=true, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@117b05b8, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0}
|
+-->FrameLayout{id=-1, visibility=VISIBLE, width=1080, height=1701, has-focus=true, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@116faf91, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=75.0, child-count=1}
|
+--->FitWindowsLinearLayout{id=2131296301, res-name=action_bar_root, visibility=VISIBLE, width=1080, height=1701, has-focus=true, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@260a52f7, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=2}
|
+---->ViewStubCompat{id=2131296312, res-name=action_mode_bar_stub, visibility=GONE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=true, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@39b6c564, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0}
|
+---->ContentFrameLayout{id=16908290, res-name=content, visibility=VISIBLE, width=1080, height=1701, has-focus=true, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@124eb3cd, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=1}
|
+----->RelativeLayout{id=-1, visibility=VISIBLE, width=1080, height=1701, has-focus=true, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@2fccd3d0, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=7}
|
+------>Toolbar{id=2131296697, res-name=toolbar, visibility=VISIBLE, width=1080, height=168, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@259fbfc9, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=3}
|
+------->AppCompatImageButton{id=-1, desc=Navigate up, visibility=VISIBLE, width=168, height=168, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=androidx.appcompat.widget.Toolbar$LayoutParams@fea16ce, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0}
|
+------->AppCompatTextView{id=-1, visibility=VISIBLE, width=433, height=81, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.appcompat.widget.Toolbar$LayoutParams@af2a9ef, tag=null, root-is-layout-requested=false, has-input-connection=false, x=180.0, y=43.0, text=fieldlist-updates, input-type=0, ime-target=false, has-links=false}
|
+------->ActionMenuView{id=-1, visibility=VISIBLE, width=0, height=168, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.appcompat.widget.Toolbar$LayoutParams@36f35cfc, tag=null, root-is-layout-requested=false, has-input-connection=false, x=1080.0, y=0.0, child-count=0}
|
+------>ProgressBar{id=2131296581, res-name=progressBar, visibility=GONE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=true, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@32db8f85, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0}
|
+------>AppCompatTextView{id=2131296570, res-name=pathtext, visibility=GONE, width=1056, height=57, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=true, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@1314d1da, tag=null, root-is-layout-requested=false, has-input-connection=false, x=12.0, y=180.0, text=Single relevance at end, input-type=0, ime-target=false, has-links=false}
|
+------>LinearLayout{id=2131296353, res-name=buttonholder, visibility=VISIBLE, width=1080, height=144, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@2d974ce8, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=1557.0, child-count=3}
|
+------->AppCompatButton{id=2131296486, res-name=jumpBeginningButton, visibility=VISIBLE, width=551, height=144, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@264a9f01, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, text=Go To Start, input-type=0, ime-target=false, has-links=false}
|
+------->AppCompatButton{id=2131296487, res-name=jumpEndButton, visibility=VISIBLE, width=529, height=144, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@5fe31a6, tag=null, root-is-layout-requested=false, has-input-connection=false, x=551.0, y=0.0, text=Go To End, input-type=0, ime-target=false, has-links=false}
|
+------->AppCompatButton{id=2131296429, res-name=exitButton, visibility=GONE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=true, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@cbfe7e7, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, text=Exit, input-type=0, ime-target=false, has-links=false}
|
+------>RecyclerView{id=2131296505, res-name=list, visibility=VISIBLE, width=1080, height=1389, has-focus=true, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=true, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@2a084f94, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=168.0, child-count=8}
|
+------->RelativeLayout{id=-1, visibility=VISIBLE, width=1080, height=205, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.recyclerview.widget.RecyclerView$LayoutParams@e412a3d, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=-67.0, child-count=3}
|
+-------->AppCompatImageView{id=2131296473, res-name=icon, visibility=VISIBLE, width=72, height=72, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@14cbc232, tag=null, root-is-layout-requested=false, has-input-connection=false, x=30.0, y=66.0}
|
+-------->AppCompatTextView{id=2131296580, res-name=primary_text, visibility=VISIBLE, width=503, height=88, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@14991b83, tag=null, root-is-layout-requested=false, has-input-connection=false, x=126.0, y=30.0, text=Cascading select, input-type=0, ime-target=false, has-links=false}
|
+-------->AppCompatTextView{id=2131296628, res-name=secondary_text, visibility=VISIBLE, width=114, height=57, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@35e4d100, tag=null, root-is-layout-requested=false, has-input-connection=false, x=126.0, y=118.0, text=Group, input-type=0, ime-target=false, has-links=false}
|
+------->RelativeLayout{id=-1, visibility=VISIBLE, width=1080, height=205, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.recyclerview.widget.RecyclerView$LayoutParams@101b2d39, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=140.0, child-count=3}
|
+-------->AppCompatImageView{id=2131296473, res-name=icon, visibility=VISIBLE, width=72, height=72, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@166fcf7e, tag=null, root-is-layout-requested=false, has-input-connection=false, x=30.0, y=66.0}
|
+-------->AppCompatTextView{id=2131296580, res-name=primary_text, visibility=VISIBLE, width=755, height=88, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@1847ecdf, tag=null, root-is-layout-requested=false, has-input-connection=false, x=126.0, y=30.0, text=Cascading select minimal, input-type=0, ime-target=false, has-links=false}
|
+-------->AppCompatTextView{id=2131296628, res-name=secondary_text, visibility=VISIBLE, width=114, height=57, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@1445fd2c, tag=null, root-is-layout-requested=false, has-input-connection=false, x=126.0, y=118.0, text=Group, input-type=0, ime-target=false, has-links=false}
|
+------->RelativeLayout{id=-1, visibility=VISIBLE, width=1080, height=205, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.recyclerview.widget.RecyclerView$LayoutParams@30f863f5, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=347.0, child-count=3}
|
+-------->AppCompatImageView{id=2131296473, res-name=icon, visibility=VISIBLE, width=72, height=72, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@cb5658a, tag=null, root-is-layout-requested=false, has-input-connection=false, x=30.0, y=66.0}
|
+-------->AppCompatTextView{id=2131296580, res-name=primary_text, visibility=VISIBLE, width=460, height=88, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@e877fb, tag=null, root-is-layout-requested=false, has-input-connection=false, x=126.0, y=30.0, text=Push off screen, input-type=0, ime-target=false, has-links=false}
|
+-------->AppCompatTextView{id=2131296628, res-name=secondary_text, visibility=VISIBLE, width=114, height=57, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@3eabc018, tag=null, root-is-layout-requested=false, has-input-connection=false, x=126.0, y=118.0, text=Group, input-type=0, ime-target=false, has-links=false}
|
+------->RelativeLayout{id=-1, visibility=VISIBLE, width=1080, height=205, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.recyclerview.widget.RecyclerView$LayoutParams@2e384a71, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=554.0, child-count=3}
|
+-------->AppCompatImageView{id=2131296473, res-name=icon, visibility=VISIBLE, width=72, height=72, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@190c5056, tag=null, root-is-layout-requested=false, has-input-connection=false, x=30.0, y=66.0}
|
+-------->AppCompatTextView{id=2131296580, res-name=primary_text, visibility=VISIBLE, width=655, height=88, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@233c98d7, tag=null, root-is-layout-requested=false, has-input-connection=false, x=126.0, y=30.0, text=Push off screen binary, input-type=0, ime-target=false, has-links=false}
|
+-------->AppCompatTextView{id=2131296628, res-name=secondary_text, visibility=VISIBLE, width=114, height=57, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@4a8c5c4, tag=null, root-is-layout-requested=false, has-input-connection=false, x=126.0, y=118.0, text=Group, input-type=0, ime-target=false, has-links=false}
|
+------->RelativeLayout{id=-1, visibility=VISIBLE, width=1080, height=205, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.recyclerview.widget.RecyclerView$LayoutParams@21b61cad, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=761.0, child-count=3}
|
+-------->AppCompatImageView{id=2131296473, res-name=icon, visibility=VISIBLE, width=72, height=72, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@32151be2, tag=null, root-is-layout-requested=false, has-input-connection=false, x=30.0, y=66.0}
|
+-------->AppCompatTextView{id=2131296580, res-name=primary_text, visibility=VISIBLE, width=402, height=88, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@3447eb73, tag=null, root-is-layout-requested=false, has-input-connection=false, x=126.0, y=30.0, text=Guidance hint, input-type=0, ime-target=false, has-links=false}
|
+-------->AppCompatTextView{id=2131296628, res-name=secondary_text, visibility=VISIBLE, width=114, height=57, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@3a4e7a30, tag=null, root-is-layout-requested=false, has-input-connection=false, x=126.0, y=118.0, text=Group, input-type=0, ime-target=false, has-links=false}
|
+------->RelativeLayout{id=-1, visibility=VISIBLE, width=1080, height=205, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.recyclerview.widget.RecyclerView$LayoutParams@2dc4d6a9, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=968.0, child-count=3}
|
+-------->AppCompatImageView{id=2131296473, res-name=icon, visibility=VISIBLE, width=72, height=72, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@25ed142e, tag=null, root-is-layout-requested=false, has-input-connection=false, x=30.0, y=66.0}
|
+-------->AppCompatTextView{id=2131296580, res-name=primary_text, visibility=VISIBLE, width=283, height=88, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@2babcbcf, tag=null, root-is-layout-requested=false, has-input-connection=false, x=126.0, y=30.0, text=Date time, input-type=0, ime-target=false, has-links=false}
|
+-------->AppCompatTextView{id=2131296628, res-name=secondary_text, visibility=VISIBLE, width=114, height=57, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@2159095c, tag=null, root-is-layout-requested=false, has-input-connection=false, x=126.0, y=118.0, text=Group, input-type=0, ime-target=false, has-links=false}
|
+------->RelativeLayout{id=-1, visibility=VISIBLE, width=1080, height=205, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.recyclerview.widget.RecyclerView$LayoutParams@2ceb3465, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=1175.0, child-count=3}
|
+-------->AppCompatImageView{id=2131296473, res-name=icon, visibility=VISIBLE, width=72, height=72, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@353a453a, tag=null, root-is-layout-requested=false, has-input-connection=false, x=30.0, y=66.0}
|
+-------->AppCompatTextView{id=2131296580, res-name=primary_text, visibility=VISIBLE, width=188, height=88, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@2fc355eb, tag=null, root-is-layout-requested=false, has-input-connection=false, x=126.0, y=30.0, text=Rating, input-type=0, ime-target=false, has-links=false}
|
+-------->AppCompatTextView{id=2131296628, res-name=secondary_text, visibility=VISIBLE, width=114, height=57, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@391b5f48, tag=null, root-is-layout-requested=false, has-input-connection=false, x=126.0, y=118.0, text=Group, input-type=0, ime-target=false, has-links=false}
|
+------->RelativeLayout{id=-1, visibility=VISIBLE, width=1080, height=205, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.recyclerview.widget.RecyclerView$LayoutParams@215fb1e1, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=1382.0, child-count=3}
|
+-------->AppCompatImageView{id=2131296473, res-name=icon, visibility=VISIBLE, width=72, height=72, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@12f77b06, tag=null, root-is-layout-requested=false, has-input-connection=false, x=30.0, y=66.0}
|
+-------->AppCompatTextView{id=2131296580, res-name=primary_text, visibility=VISIBLE, width=364, height=88, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@277f65c7, tag=null, root-is-layout-requested=false, has-input-connection=false, x=126.0, y=30.0, text=External app, input-type=0, ime-target=false, has-links=false}
|
+-------->AppCompatTextView{id=2131296628, res-name=secondary_text, visibility=VISIBLE, width=114, height=57, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@212b27f4, tag=null, root-is-layout-requested=false, has-input-connection=false, x=126.0, y=118.0, text=Group, input-type=0, ime-target=false, has-links=false}
|
+------>AppCompatTextView{id=16908292, res-name=empty, visibility=GONE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=true, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@20448b1d, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, text=Nothing available to display., input-type=0, ime-target=false, has-links=false}
|
+------>AppCompatImageView{id=-1, visibility=VISIBLE, width=1080, height=30, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@15004192, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=168.0}
|
+->View{id=16908335, res-name=statusBarBackground, visibility=VISIBLE, width=1080, height=75, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@e029763, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0}
|
+->View{id=16908336, res-name=navigationBarBackground, visibility=VISIBLE, width=1080, height=144, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@3acccf60, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=1776.0}
|
at dalvik.system.VMStack.getThreadStackTrace(Native Method)
at java.lang.Thread.getStackTrace(Thread.java:580)
at androidx.test.espresso.base.DefaultFailureHandler.getUserFriendlyError(DefaultFailureHandler.java:96)
at androidx.test.espresso.base.DefaultFailureHandler.handle(DefaultFailureHandler.java:59)
at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:322)
at androidx.test.espresso.ViewInteraction.desugaredPerform(ViewInteraction.java:178)
at androidx.test.espresso.ViewInteraction.perform(ViewInteraction.java:119)
at org.odk.collect.android.formfilling.FieldListUpdateTest.selectingADateForDateTime_ShouldChangeRelevanceOfRelatedField(FieldListUpdateTest.java:348)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
at androidx.test.rule.ActivityTestRule$ActivityStatement.evaluate(ActivityTestRule.java:531)
at androidx.test.rule.GrantPermissionRule$RequestPermissionStatement.evaluate(GrantPermissionRule.java:134)
at org.junit.rules.RunRules.evaluate(RunRules.java:20)
at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:392)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:1837)
</pre>
</details>

Update: the API level of the image the tests are run on was updated for the native library issue. Then I sent in https://github.com/opendatakit/collect/pull/3137/ in which I intended to skip `testGridSelectCompact2Appearance()` but I put an `@Ignore` on it even though it's not a test instead of commenting out the call. Regardless, I doubt that will fix anything.

<details>
<summary>Stack trace for the `AllWidgetsTest`</summary>
<pre>
androidx.test.espresso.AppNotIdleException: Looped for 1486 iterations over 60 SECONDS. The following Idle Conditions failed .
at dalvik.system.VMStack.getThreadStackTrace(Native Method)
at java.lang.Thread.getStackTrace(Thread.java:580)
at androidx.test.espresso.base.DefaultFailureHandler.getUserFriendlyError(DefaultFailureHandler.java:96)
at androidx.test.espresso.base.DefaultFailureHandler.handle(DefaultFailureHandler.java:59)
at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:322)
at androidx.test.espresso.ViewInteraction.desugaredPerform(ViewInteraction.java:178)
at androidx.test.espresso.ViewInteraction.perform(ViewInteraction.java:119)
at org.odk.collect.android.AllWidgetsFormTest.testGridSelectQuickCompact2Appearance(AllWidgetsFormTest.java:859)
at org.odk.collect.android.AllWidgetsFormTest.testActivityOpen(AllWidgetsFormTest.java:197)
at java.lang.reflect.Method.invoke(Native Method)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
at androidx.test.internal.runner.junit4.statement.RunBefores.evaluate(RunBefores.java:80)
at org.mockito.internal.junit.JUnitRule$1.evaluateSafely(JUnitRule.java:52)
at org.mockito.internal.junit.JUnitRule$1.evaluate(JUnitRule.java:43)
at androidx.test.rule.ActivityTestRule$ActivityStatement.evaluate(ActivityTestRule.java:531)
at androidx.test.rule.GrantPermissionRule$RequestPermissionStatement.evaluate(GrantPermissionRule.java:134)
at org.junit.rules.RunRules.evaluate(RunRules.java:20)
at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
at tools.fastlane.screengrab.locale.LocaleTestRule$1.evaluate(LocaleTestRule.java:32)
at org.junit.rules.RunRules.evaluate(RunRules.java:20)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:392)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:1879)
</pre>
</details>

<details>
<summary>Stack trace for the field list update test</summary>
<pre>
androidx.test.espresso.PerformException: Error performing 'single click' on view '(is displayed on the screen to the user and with text: is "Cascading select minimal")'.
at androidx.test.espresso.PerformException$Builder.build(PerformException.java:86)
at androidx.test.espresso.base.DefaultFailureHandler.getUserFriendlyError(DefaultFailureHandler.java:87)
at androidx.test.espresso.base.DefaultFailureHandler.handle(DefaultFailureHandler.java:59)
at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:322)
at androidx.test.espresso.ViewInteraction.desugaredPerform(ViewInteraction.java:178)
at androidx.test.espresso.ViewInteraction.perform(ViewInteraction.java:119)
at org.odk.collect.android.formentry.FieldListUpdateTest.jumpToGroupWithText(FieldListUpdateTest.java:390)
at org.odk.collect.android.formentry.FieldListUpdateTest.selectionChangeAtOneCascadeLevelWithMinimalAppearance_ShouldUpdateNextLevels(FieldListUpdateTest.java:259)
at java.lang.reflect.Method.invoke(Native Method)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
at androidx.test.rule.ActivityTestRule$ActivityStatement.evaluate(ActivityTestRule.java:531)
at androidx.test.rule.GrantPermissionRule$RequestPermissionStatement.evaluate(GrantPermissionRule.java:134)
at org.junit.rules.RunRules.evaluate(RunRules.java:20)
at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:392)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:1879)
Caused by: java.lang.RuntimeException: Action will not be performed because the target view does not match one or more of the following constraints:
at least 90 percent of the view's area is displayed to the user.
Target view: "AppCompatTextView{id=2131296580, res-name=primary_text, visibility=VISIBLE, width=755, height=88, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@29c760a, tag=null, root-is-layout-requested=false, has-input-connection=false, x=126.0, y=30.0, text=Cascading select minimal, input-type=0, ime-target=false, has-links=false}"
at androidx.test.espresso.ViewInteraction.doPerform(ViewInteraction.java:252)
at androidx.test.espresso.ViewInteraction.access$100(ViewInteraction.java:65)
at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:158)
at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:155)
at java.util.concurrent.FutureTask.run(FutureTask.java:237)
at android.os.Handler.handleCallback(Handler.java:739)
at android.os.Handler.dispatchMessage(Handler.java:95)
at android.os.Looper.loop(Looper.java:148)
at android.app.ActivityThread.main(ActivityThread.java:5417)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
</pre>
</details>

Ok so for the three failing tests remaining in `FieldListUpdateTest` here are screenshots and the accompanying fail:

### selectingADateForDateTime_ShouldChangeRelevanceOfRelatedField

![2019-06-13_11 27 31 790296_yxjZ_Pixel2-27-en-portrait_artifacts_sdcard_org odk collect android_screengrab_en_images_screenshots_selectingADateForDateTime_ShouldChangeRelevanceOfRelatedField_1560421869581](https://user-images.githubusercontent.com/556280/59435977-a6af1f80-8de6-11e9-963b-32fc6570420b.png)

```java
androidx.test.espresso.NoMatchingViewException: No views in hierarchy found matching: with string from resource id: <2131755587>[ok] value: OK
```

### selectingARating_ShouldChangeRelevanceOfRelatedField

![2019-06-13_11 27 31 790296_yxjZ_Pixel2-27-en-portrait_artifacts_sdcard_org odk collect android_screengrab_en_images_screenshots_selectingARating_ShouldChangeRelevanceOfRelatedField_7_1560421907239](https://user-images.githubusercontent.com/556280/59435981-a878e300-8de6-11e9-8d41-3a8aad40929c.png)

```java
 Caused by: java.lang.RuntimeException: Action will not be performed because the target view does not match one or more of the following constraints:
at least 90 percent of the view's area is displayed to the user.
Target view: "ImageButton{id=7, visibility=VISIBLE, width=132, height=132, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=android.widget.GridLayout$LayoutParams@7206d660, tag=null, root-is-layout-requested=false, has-input-connection=false, x=924.0, y=0.0}" 
```

### selectionChangeAtOneCascadeLevelWithMinimalAppearance_ShouldUpdateNextLevels

![2019-06-13_11 27 31 790296_yxjZ_Pixel2-27-en-portrait_artifacts_sdcard_org odk collect android_screengrab_en_images_screenshots_selectionChangeAtOneCascadeLevelWithMinimalAppearance_ShouldUpdateNextLevels_1560421910205](https://user-images.githubusercontent.com/556280/59435984-a9aa1000-8de6-11e9-8edc-dc4a10539412.png)

```java
 Caused by: java.lang.RuntimeException: Action will not be performed because the target view does not match one or more of the following constraints:
at least 90 percent of the view's area is displayed to the user.
Target view: "AppCompatTextView{id=2131296580, res-name=primary_text, visibility=VISIBLE, width=698, height=82, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@191f2ef, tag=null, root-is-layout-requested=false, has-input-connection=false, x=116.0, y=28.0, text=Cascading select minimal, input-type=0, ime-target=false, has-links=false}" 
```

--- a/collect_app/src/androidTest/java/org/odk/collect/android/formentry/FieldListUpdateTest.java
+++ b/collect_app/src/androidTest/java/org/odk/collect/android/formentry/FieldListUpdateTest.java
@@ -346,7 +346,7 @@ public class FieldListUpdateTest {
         onView(withText("Target12")).check(doesNotExist());
 
         onView(withText(R.string.select_date)).perform(click());
-        onView(withText(R.string.ok)).perform(click());
+        onView(withId(android.R.id.button1)).perform(click());
 
         onView(withText("Target12")).check(matches(isDisplayed()));
     }

